dask/array/linalg.py

```diff
@@ -1410,11 +1410,11 @@ def lstsq(a, b):
 
     # singular
     sname = "lstsq-singular-" + token
-    rt = r.T
+    rt = r.T.conj()
     sdsk = {
         (sname, 0): (
             _sort_decreasing,
-            (np.sqrt, (np.linalg.eigvals, (np.dot, (rt.name, 0, 0), (r.name, 0, 0)))),
+            (np.sqrt, (np.linalg.eigvalsh, (np.dot, (rt.name, 0, 0), (r.name, 0, 0)))),
```

Comment:
This is saying that `r . r^H` is hermitian, and so we can use the (generally) faster function, right? Note that `eigvalsh` produces sorted eigenvalues (lowest first), so maybe can drop `_sort_decreasing` in favour of `reversed` (or `lambda x: x[::-1]`).

Reply:
That's right - and also guaranteeing that ``s`` will be real. Good point about ``_sort_reversed``, I've made that change.

Reply:
Actually, sorry to be a nuisance, maybe a function is better than a lambda, since it will serialise much more compactly. So then just removing the `.sort()` from what was before would do, and rename it to `_reverse`

Reply:
Oh yeah oh course - updating that now.